### ISSUE TITLE
Unable to register the session service for stubbing

### ISSUE THREAD

**Issue by rmmmp** [Vestorly/torii#396]
Ever since I upgraded from Ember 2.16 to 2.18, my stubs for `session` no longer works. Here's my setup..

In my `app/config/environment.js`

```javascript
torii: { sessionServiceName: 'session' }
```

I'm using the [Ember QUnit Simplification](http://rwjblue.com/2017/10/23/ember-qunit-simplication/). In my integration test I have this

```javascript
import { module } from 'qunit';
import { setupRenderingTest } from 'ember-qunit';

module('Integration | Component | foo bar', function(hooks) {
  setupRenderingTest(hooks);

  hooks.beforeEach(function() {
    this.owner.register('service:session', Service.extend({ foobar: true });

    // This returns undefined
    console.log(this.owner.lookup('service:session').get('foobar'));
  });
});
```

However, if I register a different service, like say `service:something-else` it would work. Also, if I set `sessionServiceName` to something else other than `session` the `this.owner.register('service:session')` would now work. I'm not sure if the problem is with torii or not.

**Comment by rmmmp** [comment]
I've figured out what **change** caused the problem though I'm not sure what the problem really is. It could probably be because torii's session isn't a typical service in that it's defined during runtime.

Anyways, so when I updated to Ember CLI 2.18.0, it updated my `tests/test-helper.js` from

```javascript
import resolver from './helpers/resolver';
import {
  setResolver
} from 'ember-qunit';
import { start } from 'ember-cli-qunit';

setResolver(resolver);
start();
```

To this

```javascript
import Application from '../app';
import config from '../config/environment';
import { setApplication } from '@ember/test-helpers';
import { start } from 'ember-qunit';

setApplication(Application.create(config.APP));

start();
```

@rwjblue, we talked about this briefly in slack. I didn't file an issue under `ember-test-helpers` since it's only affecting the `session` service in `torii`.

**Comment by rwjblue** [comment]
@rmmmp - Can you push up a demo repo to make it easier for me to dig around?

**Comment by rmmmp** [comment]
Sure. I'll set one up when I get in front of my laptop.

**Comment by rmmmp** [comment]
@rwjblue, [here's](https://github.com/rmmmp/torii-problem) the repo. You can find my tests in the `foo-bar` integration test. I've added context there and some additional findings with `store.findRecord()` causing things to not work too.

TL;DR

- Having a `store.findRecord()` run before registering a `service:session` causes the problem.
- Only the torii session service is affected.
- Using the old testing format works
- Using the old resolver as mentioned [here](https://github.com/Vestorly/torii/issues/396#issuecomment-355831726)

**Comment by mjuniper** [comment]
We are running into this issue too. Is this something that can/will be fixed?

**Comment by rajneeshnagarro** [comment]
Can you try the following?

```
test('my test', function(assert) {
    const owner = Ember.getOwner(this);
    // Your code with owner to get the service
    owner.lookup(‘service:session’)
}); 

```
